spot-ml/src/main/scala/org/apache/spot/proxy/ProxySuspiciousConnectsModel.scala

```diff
@@ -114,26 +116,50 @@ object ProxySuspiciousConnectsModel {
     val agentCuts =
       Quantiles.computeQuintiles(df.select(UserAgent).rdd.map({ case Row(agent: String) => agentToCountBC.value(agent) }))
 
-    val docWordCount: RDD[SpotLDACInput] =
-      getIPWordCounts(sparkContext, sqlContext, logger, df, config.scoresFile, config.duplicationFactor, agentToCount, timeCuts, entropyCuts, agentCuts)
-
-
-    val SpotLDACOutput(documentResults, wordResults) = SpotLDACWrapper.runLDA(docWordCount,
-      config.modelFile,
-      config.topicDocumentFile,
-      config.topicWordFile,
-      config.mpiPreparationCmd,
-      config.mpiCmd,
-      config.mpiProcessCount,
-      config.topicCount,
-      config.localPath,
-      config.ldaPath,
-      config.localUser,
-      config.analysis,
-      config.nodes,
-      config.ldaPRGSeed)
-
-    new ProxySuspiciousConnectsModel(config.topicCount, documentResults, wordResults, timeCuts, entropyCuts, agentCuts)
+    if (config.ldaImplementation == "LDAC") {
+      val docWordCount: RDD[SpotLDACInput] =
+        getIPWordCounts(sparkContext, sqlContext, logger, df, config.scoresFile, config.duplicationFactor, agentToCount, timeCuts, entropyCuts, agentCuts)
+
+
+      val SpotLDACOutput(documentResults, wordResults) = SpotLDACWrapper.runLDA(docWordCount,
```

Comment:
it seems like we could push this if-check inside a new routine (called LDA) and that way we would have the if-check in one place and not in all three SuspiciousConnectsModels